ibis/expr/types/arrays.py

```diff
@@ -504,15 +543,20 @@ def filter(
         │ []                            │
         └───────────────────────────────┘
         """
-        name = next(iter(inspect.signature(predicate).parameters.keys()))
+        if isinstance(predicate, Deferred):
+            name = "_"
```

Comment:
```suggestion
            name = str(predicate)
```

Reply:
If I do this, I get this compile error:

```
self = <duckdb_engine.ConnectionWrapper object at 0x1409dcb90>
statement = 'SELECT t0.a \nFROM (SELECT list_apply(t1.a, (__ibis_param_(_ + 1)__) -> __ibis_param_(_ + 1)__ + CAST(? AS TinyInt)) AS a \nFROM ibis_pandas_memtable_x2fpeyfflncsdl2mrrugktbgjq AS t1) AS t0'
parameters = (1,), context = None

    def execute(
        self,
        statement: str,
        parameters: Optional[Tuple] = None,
        context: Optional[Any] = None,
    ) -> None:
        try:
            if statement.lower() == "commit":  # this is largely for ipython-sql
                self.__c.commit()
            elif statement.lower() in (
                "register",
                "register(?, ?)",
                "register($1, $2)",
            ):
                assert parameters and len(parameters) == 2, parameters
                view_name, df = parameters
                self.__c.register(view_name, df)
            elif parameters is None:
                self.__c.execute(statement)
            else:
>               self.__c.execute(statement, parameters)
E               sqlalchemy.exc.ProgrammingError: (duckdb.duckdb.ParserException) Parser Error: syntax error at or near "__"
E               LINE 2: ...
E               FROM (SELECT list_apply(t1.a, (__ibis_param_(_ + 1)__) -> __ibis_param_(_ + 1...
E                                                                 ^
E               [SQL: SELECT t0.a 
E               FROM (SELECT list_apply(t1.a, (__ibis_param_(_ + 1)__) -> __ibis_param_(_ + 1)__ + CAST(? AS TinyInt)) AS a 
E               FROM ibis_pandas_memtable_x2fpeyfflncsdl2mrrugktbgjq AS t1) AS t0]
E               [parameters: (1,)]
E               (Background on this error at: https://sqlalche.me/e/14/f405)

.venv/lib/python3.11/site-packages/duckdb_engine/__init__.py:160: ProgrammingError
```

Reply:
Ah, probably needs quoting. You can keep it as is and I will fix this on `the-epic-split`.

Reply:
We'll leave this as is for now.